executor/builder.go

```diff
@@ -230,6 +233,33 @@ func (b *executorBuilder) buildCheckTable(v *plan.CheckTable) Executor {
 	return e
 }
 
+func (b *executorBuilder) buildCheckIndexRange(v *plan.CheckIndexRange) Executor {
+	dbName := model.NewCIStr(b.ctx.GetSessionVars().CurrentDB)
+	tb, err := b.is.TableByName(dbName, v.Table.Name)
+	if err != nil {
+		b.err = errors.Trace(err)
+		return nil
+	}
+	e := &CheckIndexRangeExec{
+		baseExecutor: newBaseExecutor(b.ctx, v.Schema(), v.ExplainID()),
+		handleRanges: v.HandleRanges,
+		table:        tb,
+		is:           b.is,
+	}
+	idxName := strings.ToLower(v.IndexName)
+	for _, idx := range tb.Indices() {
+		if idx.Meta().Name.L == idxName {
```

Comment:
Why not put it to planbuilder.go?

Reply:
I think either way is ok.

Reply:
That traversal is used to build schema, if we reduce the traversal we'll do too many things in a traversal.